core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestResourceFitness.kt

```diff
@@ -58,43 +70,100 @@ class RestResourceFitness : AbstractRestFitness<RestIndividual>() {
+                    So 2 options: (1) always reset before calling SUT, or (2) always reset after call in which external
+                    setups were used.
+                 */
+                externalServiceHandler.resetServedRequests()
+
+                val externalServiceActions = a.getExternalServiceActions()
+
+                externalServiceActions.forEach { it.resetActive() }
+
+                externalServiceActions.filter { it.active }
+                    .filterIsInstance<HttpExternalServiceAction>()
+                    .forEach {
+                        // TODO: Handling WireMock for ExternalServiceActions should be generalised
+                        //  to facilitate other cases such as RPC and GraphQL
+                        it.buildResponse()
+                    }
+
+                val restCallAction = a.getMainAction()
 
                 //TODO handling of inputVariables
-                registerNewAction(a, indexOfAction)
+                registerNewAction(restCallAction, indexOfAction)
 
                 val ok: Boolean
 
-                if (a is RestCallAction) {
-                    ok = handleRestCall(a, actionResults, chainState, cookies, tokens)
+                if (restCallAction is RestCallAction) {
+                    ok = handleRestCall(restCallAction, actionResults, chainState, cookies, tokens)
                     // update creation of resources regarding response status
                     val restActionResult = actionResults.filterIsInstance<RestCallResult>()[indexOfAction]
-                    call.getResourceNode().confirmFailureCreationByPost(call, a, restActionResult)
+                    call.getResourceNode().confirmFailureCreationByPost(call, restCallAction, restActionResult)
                     restActionResult.stopping = !ok
                 } else {
-                    throw IllegalStateException("Cannot handle: ${a.javaClass}")
+                    throw IllegalStateException("Cannot handle: ${restCallAction.javaClass}")
+                }
+
+                // get visited wiremock instances
+                val requestedExternalServiceRequests = externalServiceHandler.getAllServedExternalServiceRequests()
+                if (requestedExternalServiceRequests.isNotEmpty()) {
+                    fv.registerExternalServiceRequest(
+                        indexOfAction,
+                        requestedExternalServiceRequests
+                    )
+                }
+
+                // Mark the existing external service actions as used if there is call
+                //  made based on the absolute URL. Otherwise, mark it not used
+                requestedExternalServiceRequests.forEach { request ->
+                    externalServiceActions.filterIsInstance<HttpExternalServiceAction>().forEach { action ->
+                        if (action.request.absoluteURL == request.absoluteURL) {
+                            action.confirmUsed()
+                        } else {
```

Comment:
the if-else logic here is incorrect. you need to check the existing action regarding all of `requestedExternalServiceRequests` not one of them, eg, the existing action does not appear in any `requestedExternalServiceRequests`, we confirmed it `notUsed`

In addition, times need to be checked. Imagine that there exist two external service actions, but only one is accessed. then need to mark the first one `confirmUsed` and the second one `confirmNotUsed`